app/models/components/course/forums_ability_component.rb

```diff
@@ -50,6 +51,10 @@ def allow_student_reply_unlocked_topics
     cannot :reply, Course::Forum::Topic, topic_all_course_users_hash.reverse_merge(locked: true)
   end
 
+  def allow_student_resolve_own_topics
+    can :set_resolved, Course::Forum::Topic, creator_id: user.id
```

Comment:
Will need to add the ability spec for this. I was wondering if we should instead just place `can :resolve` instead of `set_resolved`?

Reply:
Do you mean this? https://github.com/Coursemology/coursemology2/pull/2486/files#diff-0869b1080504a191d53fb2aa8960b980R36

Reply:
Yup 👍 My bad, was reviewing commit by commit >_<

Reply:
I think it needs to follow the controller action, which I named `set_resolved` to follow the existing `set_locked` and `set_hidden` actions.

Reply:
If the actions differ, then we can call `authorize! :resolve, @topic` to do the authorization. This one not very important, up to you whether you want to fix.